Differentiate between production and dev environments

p-galligan opened the pull request:
Adds a second config file for development purposes, and tweaks the build script.

So, working off the assumption that we can set `jekyll_env` on the server side, all we'd need to do is set it to `production` on the production server and it'll run off of `_config.yml` settings.

Otherwise, any other environment setting will use the `_config_dev.yml` settings when building. The settings loaded from the later config will override settings in the earlier one.

LMK if you think this won't work for our deployment.

Commit:
add jekyll env

Adds jekyll production environment and site url to config

Commit:
differentiate configs

Build based on different configs depending on environment variables.

Commit:
slight update to build

slightly updates build commands

Commit:
update travis file

Updates travis file to include jekyll env

Commit:
printing out in build

checking travis by printing to see what's running

helrond commented on script/build:
```diff
@@ -3,7 +3,13 @@ set -e # halt script on error
 python script/make_pages.py
 
-bundle exec jekyll build
+if JEKYLL_ENV=production; then
```
```suggestion
if [ "$JEKYLL_ENV" = "production"]; then
```

helrond commented on script/build:
```diff
@@ -3,7 +3,13 @@ set -e # halt script on error
+  bundle exec jekyll build --config _config.yml
+  echo "production"
+else
+  bundle exec jekyll build --config _config.yml,_config_dev.yml
```
```suggestion
  bundle exec jekyll build --config _config_dev.yml
```

p-galligan committed:
Update script/build

Co-authored-by: Hillel Arnold <helrond@hotmail.com>

p-galligan committed:
Update script/build

Co-authored-by: Hillel Arnold <helrond@hotmail.com>

helrond commented on script/build:
```diff
@@ -3,7 +3,13 @@ set -e # halt script on error
 python script/make_pages.py
 
-bundle exec jekyll build
+if [ "$JEKYLL_ENV" = "production"]; then
```
```suggestion
if [ "$JEKYLL_ENV" = "production" ]; then
```

helrond committed:
Update script/build

helrond commented:
I think this will work. I'm wondering, though, about how wise it is to have two config files with almost identical values. If you pass multiple configs, do those values get merged? Would it be possible to have a _config.yml that has all the base values and then a _config.prod.yml that has ONLY the site.url value?

p-galligan commented:
@helrond As far as I can tell. I think your suggestion would work. "Settings in later files override settings in earlier files." So I _think_ that it should work to overwrite, and duplication isn't necessary either one way or another.

helrond commented:
OK, so let's try that then?

p-galligan commented:
Unfortunately it doesn't selectively overwrite... I just tested locally and it needed a value for every variable in both configs.

Commit:
restructure configs

Rearrange configs to default to development and then just overwrite `url` for production.

helrond commented:
Maybe just get rid of the `echo` lines in script/build?

p-galligan commented:
@helrond Yep, I have that committed locally, I just haven't pushed yet. Do you mind if I add the comment to `make_pages` and the readme change here as well?

helrond commented:
That makes sense - let's get that all in this PR.

Commit:
Build against python 2

Adds a shebang to the build script to specify python 2.7. Updates `make_pages.py` and readme to document why we made this decision.
